### ISSUE TITLE
awesome throwing out errors when changing keyboard layout via xmodmap and xcape

### ISSUE THREAD

**Issue by yilkalargaw** [awesomeWM/awesome#2861]
Output of `awesome --version`:
awesome v4.2 (Human after all)
 • Compiled against Lua 5.3.4 (running with Lua 5.3)
 • D-Bus support: ✔
 • execinfo support: ✔
 • xcb-randr version: 1.6
 • LGI version: 0.9.2

**How to reproduce the issue:**

1. install xcape

2. modify your .Xmodmap to resemble the one at [https://pastebin.com/3DjFqM41](https://pastebin.com/3DjFqM41)

3. and run 
```bash
xmodmap .myXmodmap && ps aux | grep -i '[x]cape' && killall -9 xcape; hash xcape 2>&1 >/dev/null && xcape -e 'Mode_switch=semicolon;Hyper_R=Return;Shift_L=Escape;Shift_R=Escape'  &

```

What the .Xmodmap and xcape command does is

- switch ctrl by alt and viceversa
-  use Caps lock and Enter keys as Hyper modifiers when pressed with other keys
- semicolon to serve as Mod_Switch
- semicolon, colon, and Enter to serve as themselves when pressed separately
- etc ...

**Actual result:**
spits out an error every time I press the semicolon(;) or colon(:) without combination 
```
/usr/share/awesome/lib/awful/widget/keyboardla;;yout.lua.123: attempt to concatenate a nil value (field?)
```
**Expected result:**

should be possible to type the characters without the error.

**Comment by psychon** [comment]
This should be fixed by acc012353d03ad2c1e0783e19fe5f1f1c6a4c798 / #1941 and looks to me like a duplicate of #1933. Can you test if applying that patch by hand fixes your issue? (The patch is in v4.3, in case this info helps you)

**Comment by yilkalargaw** [comment]
I just disabled the keyboard layout widget for now.
